tests/tests_common_test.go

```diff
@@ -111,6 +111,49 @@ func expectRecreateAfterDelete(res *testResource) {
 	Expect(err).ToNot(HaveOccurred())
 }
 
+func expectMetricsIncreaseAfterRestore(res *testResource) {
+	restoredCountBefore := totalRestoredTemplatesCount()
+	expectRestoreAfterUpdate(res)
+	Expect(totalRestoredTemplatesCount() - restoredCountBefore).To(Equal(1))
+}
+
+func sspOperatorReconcilingProperly() (sum int) {
+	operatorPods, operatorMetricsPort := operatorPodsWithMetricsPort()
+	for _, sspOperator := range operatorPods.Items {
+		sum += intMetricValue("ssp_operator_reconciling_properly", operatorMetricsPort, &sspOperator)
+	}
+	return
+}
+
+func totalRestoredTemplatesCount() (sum int) {
+	operatorPods, operatorMetricsPort := operatorPodsWithMetricsPort()
+	for _, sspOperator := range operatorPods.Items {
+		sum += intMetricValue("total_restored_common_templates", operatorMetricsPort, &sspOperator)
+	}
+	return
+}
+
+func operatorPodsWithMetricsPort() (core.PodList, uint16) {
+	pods := &core.PodList{}
+	err := apiClient.List(ctx, pods, client.MatchingLabels{"control-plane": "ssp-operator"})
+	Expect(err).ToNot(HaveOccurred())
+	Expect(pods.Items).ToNot(BeEmpty())
+	operatorMetricsPort, err := metricsPort(*pods)
+	Expect(err).ToNot(HaveOccurred())
+	return *pods, operatorMetricsPort
+}
+
+func metricsPort(pods core.PodList) (uint16, error) {
```

Comment:
First part done.
As for the error - I get your point, I think this is a bit different. I added another case where it can fail, and would like to see different errors if it does.